MAINT: Updated tick and category test formatting

madphysicist opened the pull request:
Rearranged tests in `lib/matplotlib/tests/test_ticker.py` and made a couple of very minor changes to `lib/matplotlib/tests/test_category.py`. Based on comment by @story645 in PRs #7482 and #7965.

This PR should probably be merged before either of the others so I can rebase them onto it and clean things up.

story645 commented on lib/matplotlib/tests/test_ticker.py:
```diff
@@ -482,82 +500,102 @@ def test_LogFormatterSciNotation():
+        assert formatter.format_pct(x, display_range) == expected
+
+
+class TestEndFormatter(object):
```
typo: EngFormatter

story645 commented on lib/matplotlib/tests/test_category.py:
```diff
@@ -122,6 +122,25 @@ def lt(tl):
+        np.array(['1', '11', '3']),
+        [b'1', b'11', b'3'],
+        np.array([b'1', b'11', b'3'])],
+    ]
```
apparently an extra one here messing up appveyor?

story645 commented:
😄 thanks!

madphysicist commented:
Thanks for looking through the PRs. I fixed up the issues you found.

madphysicist commented:
I am pretty sure that this PR is ready to go. The failing tests appear to be caused by an external source.

madphysicist commented:
No more failing tests.

madphysicist commented:
@dstansby. I've been seeing [MRG+1] a lot recently on PR titles. What does it mean?

story645 commented:
It means your PR has been approved by one reviewer (matplotlib asks for 2 approvals before commiting). http://matplotlib.org/devel/coding_guide.html#pr-review-guidelines

madphysicist commented:
Can you be the other one?

story645 commented on lib/matplotlib/tests/test_ticker.py:
```diff
@@ -482,82 +501,104 @@ def test_LogFormatterSciNotation():
+        cases, especially the case when no SI prefix is present, for values in
+        [1, 1000).
+
+        Should not raise exceptions.
```
don't need this comment.

story645 commented on lib/matplotlib/tests/test_ticker.py:
```diff
@@ -482,82 +501,104 @@ def test_LogFormatterSciNotation():
+
+        Should not raise exceptions.
+        """
+        unitless = mticker.EngFormatter()
```
these should probably be two separate tests (unitless and with unit) and possibly parameterized

story645 commented:
Sorry thought I approved. Have two comments, but if you wanna punt those to a different improve engineering test PR, let me know and I'll merge.

madphysicist commented:
@story645 I made the improvements here. I kept is as a single parametrized test, removed the extra comment and also took care of some extra whitespace in ticker.py.

madphysicist commented:
Not an actual failure, Mac test just didn't run.

QuLogic commented:
With #7973 merged, this has conflicts. When you rebase, please be sure to leave out any empty `cleanup` decorators.

story645 commented:
I think on the rebase or something, you accidentally introduced a conflict. Please resolve.

madphysicist commented:
@QuLogic Should I replace `@cleanup(style='classic')` with `@pytest.mark.style('classic')`, and make all similar changes that make sense?

QuLogic commented:
Classic is the default now, it's not necessary if that's really what's being used (and I realize I forgot to remove a couple.)

madphysicist commented:
In that case I will remove those too.

madphysicist commented:
Just to make triple sure, I should make the replacement for `'default'` though?

QuLogic commented:
That's correct.

madphysicist commented:
Done. I ended up parametrizing a couple of the tests and adding one `pytest.mark.style('default')` that I don't think was there before to make things pass. Should be good now.

madphysicist commented:
Again looks like Mac test didn't run. No actual failures.

madphysicist commented:
I've closed and reopened the PR to restart Travis and hopefully get the Mac test to run, but if you approve, this PR is ready to go.

QuLogic commented on lib/matplotlib/tests/test_ticker.py:
```diff
@@ -12,267 +12,282 @@
+                               9.441, 12.588])
+        assert_almost_equal(loc.tick_values(-7, 10), test_value)
+
+    def test_MultipleLocator_set_params(self):
```
Seems like you forgot to remove the class in this one.

QuLogic commented on lib/matplotlib/tests/test_ticker.py:
```diff
@@ -12,267 +12,282 @@
+        fmt = ax.xaxis.get_major_formatter()
+        fmt.set_locs(ax.xaxis.get_majorticklocs())
+        show_major_labels = [fmt(x) != '' for x in
+                                            ax.xaxis.get_majorticklocs()]
```
Not aligned properly; better to wrap at `for`.

QuLogic commented on lib/matplotlib/tests/test_ticker.py:
```diff
@@ -296,59 +310,60 @@ def get_view_interval(self):
+    @pytest.mark.parametrize(
+            'labelOnlyBase, exponent, locs, positions, expected', param_data)
+    @pytest.mark.parametrize('base', base_data)
+    def test_LogFormatterExponent(self, labelOnlyBase, base, exponent,
```
Class name is still here.

QuLogic commented on lib/matplotlib/tests/test_ticker.py:
```diff
@@ -296,59 +310,60 @@ def get_view_interval(self):
+
+
+class TestLogFormatterSciNotation(object):
+    testdata = [
```
Why `testdata` and not `test_data` like the others?

madphysicist replied:
I wasn't sure if it would cause any problems if I started it with `test_`.

QuLogic replied:
We'll see how the tests go, then, but I'd guess this shouldn't be a problem since it isn't callable.

madphysicist commented:
@QuLogic Made the following changes:
- Removed 2 class names in method names.
- Renamed `testdata` to `test_data`.
- Wrapped list comprehension with `for` on new line.

madphysicist commented:
Ready to go?
